arch/x86/kvm/vmx/pkvm_init.c

```diff
@@ -695,7 +695,7 @@ static __init void init_guest_state_area_from_native(struct vcpu_vmx *vmx)
 	vmcs_write32(GUEST_IDTR_LIMIT, dt.size);
 
 	/* Set MSRs */
-	vmcs_write64(GUEST_IA32_DEBUGCTL, 0);
+	vmcs_write64(GUEST_IA32_DEBUGCTL, get_debugctlmsr());
```

Comment:
As the host VMCS supports VM_EXIT_SAVE_DEBUG_CONTROLS and VM_ENTRY_LOAD_DEBUG_CONTROLS, add one more squashme patch to program the GUEST_IA32_DEBUGCTL VMCS field with the proper debug control msr value used by the host before deprivileging.

If it looks good, maybe can also squash it together with the first two patches to pkvm-v6.18.

Reply:
I'm not really familiar with this DEBUGCTL MSR, but yeah, seems like the right thing to do.